reads-utils/guess-encoding.py

```diff
@@ -46,6 +60,30 @@ def get_encodings_in_range(rmin, rmax, ranges=RANGES):
     return valid_encodings
 
 
+def heuristic_filter(valid, qual_val_counts):
+    """Apply heuristics to particular ASCII value scores
+       to try to narrow-down the encoding, beyond min/max.
+    """
+
+    if 'Illumina-1.5' in valid:
+        # 64–65: Phread+64 quality scores of 0–1 ('@'–'A')
+        #        unused in Illumina 1.5+
+        if qual_val_counts[64] > 0 or qual_val_counts[65] > 0:
+            valid.remove('Illumina-1.5')
+
+        # 66: Phread+64 quality score of 2 'B'
+        #     used by Illumina 1.5+ as QC indicator
+        elif 66 in map(operator.itemgetter(0),
+                       qual_val_counts.most_common(N_MOST_COMMON_THRESH)):
+            print("# A large number of 'B' quality scores (value 2, ASCII 66) "
+                  "were detected, which makes it likely that this encoding is "
+                  "Illumina-1.5, which has been returned as the only option.",
+                  file=sys.stderr)
+            valid = ['Illumina-1.5']
```

Comment:
I would also check that `Illumina-1.5` is in the current `valid` list before doing this.

Reply:
Thanks—will do!

Reply:
This is already done, I believe sufficiently, at the above block.

Reply:
indeed.